src/main/scala/sbtprotoc/ProtocPlugin.scala

```diff
@@ -195,4 +198,24 @@ object ProtocPlugin extends AutoPlugin with Compat {
     val extractedFiles = unpack((managedClasspath in (ProtobufConfig, key)).value.map(_.data), (PB.externalIncludePath in key).value, (streams in key).value.log)
     UnpackedDependencies((PB.externalIncludePath in key).value, extractedFiles)
   }
+
+    /**
+    * Gets a Seq[File] representing the proto sources of all the projects that the current project depends on.
+    */
+  def protocIncludeDependencies: Def.Initialize[Seq[File]] = Def.settingDyn {
+    val deps = buildDependencies.value.classpath
+
+    def getAllProjectDeps(ref: ProjectRef)(visited: Set[ProjectRef] = Set(ref)): Seq[ProjectRef] =
+      deps.getOrElse(ref, Seq.empty).map(_.project).filterNot(visited).flatMap(getAllProjectDeps(_)(visited + ref)) :+ ref
+
+    val thisProjectDeps = getAllProjectDeps(thisProjectRef.value)()
+
+    thisProjectDeps.map(ref => PB.protoSources in (ref, Compile)).foldLeft(Def.setting(Seq.empty[File])) {
```

Comment:
Shouldn't we also add PB.includePath of the dependent project? I think we also would have to remove duplicates. Can you add a test case for this?

Reply:
Yeah, I think you're right... As for duplicates, the results are filtered through a set to get distinct results.

Reply:
I see what you mean now, regarding duplicates.  Will fix.